src/base.ml

```diff
@@ -89,33 +89,40 @@ let run (title : string) (boot : boot_func option) (tick : tick_func) (s : Scree
 
         let updated_buffer = tick t s prev_buffer keys in
 
-        framebuffer_to_bigarray s updated_buffer bitmap;
-
-        match render_texture r texture s bitmap with
-        | Error (`Msg e) -> Sdl.log "Boot error: %s" e
-        | Ok () -> (
-          let exit, keys =
-          match Sdl.poll_event (Some e) with
-          | true -> (
-            match Sdl.Event.(enum (get e typ)) with
-            | `Quit -> (true, keys)
-            | `Key_down -> 
-                let key = PlatformKey.of_backend_keycode (Sdl.Event.(get e keyboard_keycode)) in
-                (false, KeyCodeSet.add key keys)
-            | `Key_up -> 
+        if updated_buffer != prev_buffer then (
+          framebuffer_to_bigarray s updated_buffer bitmap;
+          match render_texture r texture s bitmap with
+          | Error (`Msg e) -> Sdl.log "Boot error: %s" e
+          | Ok () -> ()
+        ) else if Framebuffer.is_dirty updated_buffer then (
+          framebuffer_to_bigarray s updated_buffer bitmap;
```

Comment:
This technically works, but you end up duplicating an important bit of code here in your two branches, and that means there's a risk that we update one and miss the other if we ever change `framebuffer_to_bigarray` - in software engineering this is referred to as "DRY" or "Don't Repeat Yourself".

There is actually, I think, a bug here also potentially: Surely we need to call `clear_dirty` in both cases? As even the "fresh" buffer returned from tick will still be dirty?